src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -281,11 +281,8 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                                 Kind = ex.ErrorKind
                             });
                     }
-
-                    if (IfFunction.CanCheckIfReturn(func))
-                    {
-                        Contract.Assert(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType);
-                    }
+                    
+                    Contract.Assert(result.IRContext.ResultType == node.IRContext.ResultType || result is ErrorValue || result.IRContext.ResultType is BlankType || node.IRContext.ResultType._type.IsVoid);

```

Comment:
> || node.IRContext.ResultType._type.IsVoid [](http://example.com/codeflow?start=160&length=42)

Can we avoid this?
Maybe we do need an empty  FormulaValue  that corresponds to Void? That will enforce the void type isn't being used. 
Else we maybe returning the "Real" object and using it even if the binder type is void.

Reply:
updated.